framework/src/base/FEProblemBase.C

```diff
@@ -129,6 +129,9 @@ validParams<FEProblemBase>()
                         false,
                         "This causes PETSc to error if it had to reallocate memory in the Jacobian "
                         "matrix due to not having enough nonzeros");
+  params.addParam<bool>("ignore_zeros_in_jacobian",
+                        false,
```

Comment:
This was just a question. I did not want you to change it. If you want the default to be `false`, that's fine. If setting this to `true` breaks tests, then we want to keep it as `false`...